src/Microsoft.AspNet.Mvc.TagHelpers/CacheTagHelper.cs

```diff
@@ -0,0 +1,262 @@
+        /// </summary>
+        [HtmlAttributeName(VaryByQueryAttributeName)]
+        public string VaryByQuery { get; set; }
+
+        /// <summary>
+        /// Gets or sets a comma-delimited set of route data parameters to vary the cached result by.
+        /// </summary>
+        [HtmlAttributeName(VaryByRouteAttributeName)]
+        public string VaryByRoute { get; set; }
+
+        /// <summary>
+        /// Gets or sets a comma-delimited set of cookie names to vary the cached result by.
+        /// </summary>
+        [HtmlAttributeName(VaryByCookieAttributeName)]
+        public string VaryByCookie { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value that determines if the cached result is to be varied by the Identity for the logged in
+        /// <see cref="HttpContext.User"/>.
+        /// </summary>
+        [HtmlAttributeName(VaryByUserAttributeName)]
+        public bool VaryByUser { get; set; }
+
+        /// <summary>
+        /// Gets or sets the exact <see cref="DateTimeOffset"/> the cache entry should be evicted.
+        /// </summary>
+        [HtmlAttributeName(ExpiresOnAttributeName)]
+        public DateTimeOffset? ExpiresOn { get; set; }
+
+        /// <summary>
+        /// Gets or sets the duration, from the time the cache entry was added, when it should be evicted.
+        /// </summary>
+        [HtmlAttributeName(ExpiresAfterAttributeName)]
+        public TimeSpan? ExpiresAfter { get; set; }
+
+        /// <summary>
+        /// Gets or sets the duration from last access that the cache entry should be evicted.
+        /// </summary>
+        [HtmlAttributeName(ExpiresSlidingAttributeName)]
+        public TimeSpan? ExpiresSliding { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="CachePreservationPriority"/> policy for the cache entry.
+        /// </summary>
+        [HtmlAttributeName(CachePriorityAttributeName)]
+        public CachePreservationPriority? Priority { get; set; }
+
+        /// <inheritdoc />
+        public override async Task ProcessAsync(TagHelperContext context, TagHelperOutput output)
+        {
+            var key = GenerateKey(context);
+            string result;
+            if (!MemoryCache.TryGetValue(key, out result))
+            {
+                result = await context.GetChildContentAsync();
+                MemoryCache.Set(key, cacheSetContext =>
+                {
+                    UpdateCacheContext(cacheSetContext);
+                    return result;
+                });
+            }
+
+            // Clear the contents of the "cache" element since we don't want to render it.
+            output.SupressOutput();
+
+            output.Content = result;
+        }
+
+        // Internal for unit testing
+        internal string GenerateKey(TagHelperContext context)
+        {
+            var builder = new StringBuilder(nameof(CacheTagHelper));
```

Comment:
In regards to #1808 I think the key already has been prefixed. It's probably a good idea to move the prefix to a public property so it can be reliably used in user code.